tests/unit/bokeh/core/test_json_encoder.py

```diff
@@ -41,9 +41,9 @@ def test_json_encoder():
 [null,true,false,-128,-1,0,1,128,{"type":"number","value":"nan"},{"type":"map","entries":[["key_0",{"type":"bytes","data":"dXZ3"}]]}]\
 """
 
-    assert serialize_json(rep0) == """\
-[null,true,false,-128,-1,0,1,128,{"type":"number","value":"nan"},{"type":"map","entries":[["key_0",{"type":"bytes","data":{"id":"%s"}}]]}]\
-""" % rep0.buffers[0].id
+    assert serialize_json(rep0) == f"""\
+[null,true,false,-128,-1,0,1,128,{{"type":"number","value":"nan"}},{{"type":"map","entries":[["key_0",{{"type":"bytes","data":{{"id":"{rep0.buffers[0].id}"}}}}]]}}]\
+"""
```

Comment:
FWIW I think `.format` would also be OK when it avoids `{{` escaping

Reply:
Ruff, I suppose rightfully, complains about any variant that doesn't use `{{ }}` for non-formatting. We would have to go back to %-formatting, which it will complain about as well (unless locally disabled).